foolbox/models/pytorch.py

```diff
@@ -163,24 +157,20 @@ def _loss_fn(self, image, label):
 
         image, _ = self._process_input(image)
         target = np.array([label])
-        target = torch.from_numpy(target)
-        if self.cuda:  # pragma: no cover
-            target = target.cuda()
+        target = torch.from_numpy(target).to(self.device)
         if self._old_pytorch():  # pragma: no cover
             target = Variable(target)
 
-        images = torch.from_numpy(image[None])
-        if self.cuda:  # pragma: no cover
-            images = images.cuda()
+        images = torch.from_numpy(image[None]).to(self.device)
         if self._old_pytorch():  # pragma: no cover
             images = Variable(images, volatile=True)
         predictions = self._model(images)
         ce = nn.CrossEntropyLoss()
         loss = ce(predictions, target)
         if self._old_pytorch():  # pragma: no cover
             loss = loss.data
-        if self.cuda:  # pragma: no cover
-            loss = loss.cpu()
+        if self.device != "cpu":
```

Comment:
If a tensor is already on cpu does to(device) carry any overhead? I guess it does not, in which case we could remove the check for cpu here and in other places of the script.

Reply:
Ok, I'll change it to always put it on the correct device.